core-bundle/src/EventListener/TwoFactorFrontendListener.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\PageNotFoundException;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\FrontendUser;
+use Contao\PageModel;
+use Scheb\TwoFactorBundle\Security\Authentication\Token\TwoFactorToken;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+
+class TwoFactorFrontendListener
+{
+    /** @var ContaoFramework */
+    private $framework;
+
+    /**
+     * @var ScopeMatcher
+     */
+    private $scopeMatcher;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    public function __construct(ContaoFramework $framework, ScopeMatcher $scopeMatcher, TokenStorageInterface $tokenStorage)
+    {
+        $this->framework = $framework;
+        $this->scopeMatcher = $scopeMatcher;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event): void
+    {
+        $request = $event->getRequest();
+        $token = $this->tokenStorage->getToken();
+
+        // Check if is frontend request
+        if (!$this->scopeMatcher->isFrontendRequest($request)) {
```

Comment:
Firewall stuff is only handled on the master request.

Reply:
Fixed in 300c13679.